compiler/src/dotty/tools/dotc/transform/PickleQuotes.scala

```diff
@@ -203,6 +203,9 @@ class PickleQuotes extends MacroTransform {
   private def getPicklableHoleType(tpe: Type, isStagedClasses: Symbol => Boolean)(using Context) =
     new TypeOps.AvoidMap {
       def toAvoid(tp: NamedType) = !isStagedClasses(tp.typeSymbol) && !isStaticPrefix(tp)
+      override def apply(tp: Type): Type = tp match
+        case tp: ThisType if tp.typeSymbol.isLocal => super.apply(tp.widen)
```

Comment:
In regular AvoidMaps this isn't needed:
https://github.com/lampepfl/dotty/blob/ca6a80e3c7aef6b16304df20544c41a96556c834/compiler/src/dotty/tools/dotc/core/TypeOps.scala#L478-L482
Can you add a comment explaining why we need to do something different here?

Reply:
Done